File: Chapter08/VB/VBProject_08/WebServices/4-NetRIAService/NetRIAService.Web/ResourceService.vb


Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.ComponentModel.DataAnnotations
Imports System.Linq
'Imports System.Web.Ria
'Imports System.Web.Ria.Data
'Imports System.Web.DomainServices
Imports System.Data
'Imports System.Web.DomainServices.LinqToEntities

'Namespace NetRIAService.Web
'  Implements application logic using the SilverlightResourceEntities context.
'  TODO: Add your application logic to these methods or in additional methods.
<EnableClientAccess()>
Public Class ResourceService
    Inherits LinqToEntitiesDomainService(Of SilverlightResourceEntities)

    '  TODO: Consider
    '  1. Adding parameters to this method and constraining returned results, and/or
    '  2. Adding query methods taking different parameters.
    Public Function GetResource() As IQueryable(Of Resource)
        Return Me.Context.Resource
    End Function
End Class   '   ResourceService
'End Namespace
' ..\Project_08\WebServices\4-NetRIAService\NetRIAService.Web\ResourceService.cs

File: Chapter08/C#/Project/WebServices/4-NetRIAService/NetRIAService.Web/ResourceService.cs

namespace NetRIAService.Web
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.ComponentModel.DataAnnotations;
    using System.Linq;
    using System.Web.Ria;
    using System.Web.Ria.Data;
    using System.Web.DomainServices;
    using System.Data;
    using System.Web.DomainServices.LinqToEntities;


    // Implements application logic using the SilverlightResourceEntities context.
    // TODO: Add your application logic to these methods or in additional methods.
    [EnableClientAccess()]
    public class ResourceService : LinqToEntitiesDomainService<SilverlightResourceEntities>
    {

        // TODO: Consider
        // 1. Adding parameters to this method and constraining returned results, and/or
        // 2. Adding query methods taking different parameters.
        public IQueryable<Resource> GetResource()
        {
            return this.Context.Resource;
        }
    }
}


